crates/ruff/src/rules/pylint/rules/redefined_loop_name.rs

```diff
@@ -0,0 +1,325 @@
+            ctx: ExprContext::Store,
+            value,
+            ..
+        } => Box::new(iter::once(&**value)),
+        ExprKind::Name {
+            ctx: ExprContext::Store,
+            id,
+            ..
+        } => {
+            // Ignore dummy variables.
+            if dummy_variable_rgx.is_match(id) {
+                Box::new(iter::empty())
+            } else {
+                Box::new(iter::once(expr))
+            }
+        }
+        ExprKind::List {
+            ctx: ExprContext::Store,
+            elts,
+            ..
+        } => Box::new(
+            elts.iter()
+                .flat_map(|elt| assignment_targets_from_expr(elt, dummy_variable_rgx)),
+        ),
+        ExprKind::Tuple {
+            ctx: ExprContext::Store,
+            elts,
+            ..
+        } => Box::new(
+            elts.iter()
+                .flat_map(|elt| assignment_targets_from_expr(elt, dummy_variable_rgx)),
+        ),
+        _ => Box::new(iter::empty()),
+    }
+}
+
+fn assignment_targets_from_with_items<'a, U>(
+    items: &'a [Withitem<U>],
+    dummy_variable_rgx: &'a HashableRegex,
+) -> impl Iterator<Item = &'a Expr<U>> + 'a {
+    items
+        .iter()
+        .filter_map(|item| {
+            item.optional_vars
+                .as_ref()
+                .map(|expr| assignment_targets_from_expr(&**expr, dummy_variable_rgx))
+        })
+        .flatten()
+}
+
+fn assignment_targets_from_assign_targets<'a, U>(
+    targets: &'a [Expr<U>],
+    dummy_variable_rgx: &'a HashableRegex,
+) -> impl Iterator<Item = &'a Expr<U>> + 'a {
+    targets
+        .iter()
+        .flat_map(|target| assignment_targets_from_expr(target, dummy_variable_rgx))
+}
+
+/// PLW2901
+pub fn redefined_loop_name<'a, 'b>(checker: &'a mut Checker<'b>, node: &Node<'b>) {
+    let (outer_assignment_targets, inner_assignment_targets) = match node {
+        Node::Stmt(stmt) => match &stmt.node {
+            // With.
+            StmtKind::With { items, body, .. } => {
+                let outer_assignment_targets: Vec<ExprWithBindingKind<'a>> =
+                    assignment_targets_from_with_items(items, &checker.settings.dummy_variable_rgx)
+                        .map(|expr| ExprWithBindingKind {
+                            expr,
+                            binding_kind: BindingKind::With,
+                        })
+                        .collect();
+                let mut visitor = InnerForWithAssignTargetsVisitor {
+                    dummy_variable_rgx: &checker.settings.dummy_variable_rgx,
+                    assignment_targets: vec![],
+                };
+                for stmt in body {
+                    visitor.visit_stmt(stmt);
+                }
+                (outer_assignment_targets, visitor.assignment_targets)
+            }
+            // For and async for.
+            StmtKind::For {
+                target,
+                body,
+                iter: _,
+                orelse: _,
```

Comment:
I think preferable to just omit these if you're already using `..`.

Reply:
Done.